src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -103,6 +108,22 @@ public static boolean isStringSafe(@CheckForNull String input, @CheckForNull Env
                 : true;
     }
 
+    /** Restrict stored arguments to a reasonable subset of types so we don't retain totally arbitrary objects
+     *  in memory. Generally we aim to allow storing anything that maps correctly to an {@link UninstantiatedDescribable}
+     *  But we may allow a few extras doesn't address, if they're safe to store.
+     */
+    boolean isStoreableType(Object ob) {
+        if (ob == null) {
+            return true;
+        } else if (ob instanceof CharSequence || ob instanceof Number || ob instanceof Boolean
```

Comment:
I'm really not 100% on which types are legal here, so I'm opting to include a few extra that seem to be mentioned within the Structs plugin -- and some things that we know are safe and useful to store/visualize in some fashion.

Reply:
Note that we're putting this in most-likely order after the nullcheck (most steps take Strings, booleans, or numbers).

Reply:
How common do we think enums/arrays are? Maybe it would make sense to move `c.isPrimitive() || c.isEnum() || c.isArray()` up before the `instanceof` checks, although I would be surprised if it made a difference in practice.